XLS-30/README.md

```diff
@@ -0,0 +1,1254 @@
+| `tfTwoAsset`      | 0x00100000  |   If set, it indicates `Amount` and `Amount2` parameter combination and may optionally include `LPTokenOut` paramater|
+ | `tfOneAssetLPToken`      | 0x00200000 |   If set, it indicates `Amount` and `LPTokenOut` parameter combination|
+ | `tfLimitLPToken`      | 0x00400000  |   If set, it indicates `Amount` and `EPrice` parameter combination|
+ | `tfWithdrawAll`      | 0x00020000  |   If set, it indicates proportional withdrawal of all `LPTokens`|
+| `tfTwoAssetIfEmpty`| 0x00800000  |   If set, it indicates that this deposit can only be submitted on an empty state AMM (LPTokens == 0)|
+
+If `tfTwoAssetIfEmpty` is set, both amounts have to be specified  and deposited into AMM as is. It is sort of like `AMMCreate` in an empty AMM state.
+ 
+ Following are the recommended valid combinations. Other invalid combinations may result in the failure of transaction.
+
+- `LPTokenOut`, `[Amount]`, `[Amount2]`
+- `Amount`, `[LPTokenOut]`
+- `Amount` , `Amount2`, `[LPTokenOut]`
+- `Amount` and `LPTokenOut`
+- `Amount` and `EPrice` 
+
+Details for above combinations:
+
+1. Fields specified: `LPTokenOut`, `[Amount]`, `[Amount2]` and Flag: `tfLPToken`
+
+Such a transaction assumes proportional deposit of pools assets in exchange for the specified amount of `LPTokenOut` of the AMM instance. `Amount` and `Amount2`, if included, have to be provided both and specify minimum deposit amounts for each asset.
+Deposit fails if the min deposit condition is not met
+
+2. Fields specified: `Amount`, `[LPTokenOut]` and Flag: `tfSingleAsset`
+
+Such a transaction assumes single asset deposit of the amount of asset specified by `Amount`. This is essentially an _equal_ asset deposit and a _swap_.
+
+If the asset to be deposited is a token, specifying the `value` field is required, else the transaction will fail.
+
+`LPTokenOut`, if included, specified minimum `LPTokens` amount that the user receives, else the transaction will fail.
+
+3. Fields specified: `Amount`, `Amount2`, `[LPTokenOut]` and Flag: `tfTwoAsset`
+
+Such a transaction assumes proportional deposit of pool assets with the constraints on the maximum amount of each asset that the trader is willing to deposit.
+
+`LPTokenOut`, if included, specified minimum `LPTokens` amount that the user receives, else the transaction will fail.
+
+4. Fields specified: `Amount` and `LPTokenOut` and Flag: `tfOneAssetLPToken`
+
+Such a transaction assumes that a single asset `Amount` is deposited to obtain some share of the AMM instance's pools represented by amount of `LPTokenOut`. Since adding liquidity to the pool with one asset changes the ratio of the assets in the two pools, thus changing the relative pricing, trading fee is charged only on the amount of the deposited asset that causes this change.
+
+5. Fields specified: `Amount` and `EPrice` and Flag: `tfLimitLPToken`
+
```

Comment:
Add 6. `Amount`, `Amount2`, [`TradingFee`] and Flag: `tfTwoAssetIfEmpty`.

Reply:
Added